python/pylibcudf/pylibcudf/scalar.pyx

```diff
@@ -79,3 +91,31 @@ cdef class Scalar:
         s.c_obj.swap(libcudf_scalar)
         s._data_type = DataType.from_libcudf(s.get().type())
         return s
+
+    @classmethod
+    def from_py(cls, py_val):
+        """Convert a Python standard library object to a Scalar.
+        """
+        cdef DataType dtype
+        if isinstance(py_val, py_bool):
+            dtype = DataType(type_id.BOOL8)
+            c_val = make_numeric_scalar(dtype.c_obj)
+            (<numeric_scalar[cbool]*>c_val.get()).set_value(py_val)
+        elif isinstance(py_val, int):
+            dtype = DataType(type_id.INT64)
+            c_val = make_numeric_scalar(dtype.c_obj)
+            (<numeric_scalar[int64_t]*>c_val.get()).set_value(py_val)
+        elif isinstance(py_val, float):
+            dtype = DataType(type_id.FLOAT64)
+            c_val = make_numeric_scalar(dtype.c_obj)
+            (<numeric_scalar[double]*>c_val.get()).set_value(py_val)
+        elif isinstance(py_val, str):
+            dtype = DataType(type_id.STRING)
+            c_val = make_string_scalar(py_val.encode())
+        else:
+            raise NotImplementedError(f"{type(py_val).__name__} is not supported.")
```

Comment:
Can we rewrite this pattern with singledispatch? I anticipate this cascade eventually getting arbitrarily large and we'll start paying the price of failing through earlier cases to get to later ones. That also gives you the benefit of (since each overload is a separate function) being able to cdef the c_val type accordingly in each case.

Reply:
Good idea, how would you handle creating the final Scalar with multiple dispatched functions? I'm referring to this logic
```
        cdef Scalar s = Scalar.__new__(Scalar)
        s.c_obj.swap(c_val)
        s._data_type = dtype
        return s
```

Reply:
Oh would this just go in a separate method so it can be used in each dispatched method?

Reply:
Eg.
```
@staticmethod
def _finalize_scalar(c_obj, dtype):
        cdef Scalar s = Scalar.__new__(Scalar)
        s.c_obj.swap(c_obj)
        s._data_type = dtype
        return s
```

Reply:
That's a good question. Yes I think you would have to put that in a separate function. You would probably want to make it a cdef function with strongly typed arguments because you need `c_obj` to be a typed cudf::scalar in order to be able to call methods like `s.c_obj.swap`.

Reply:
Sure, was able to use a `singledispatch` pattern here. Let me know if it's what you had in mind